src/torchmetrics/functional/segmentation/mean_iou.py

```diff
@@ -0,0 +1,233 @@
+        label[label == 254] = 255
+
+    mask = label != ignore_index
+    pred_label = pred_label[mask]
+    label = label[mask]
+
+    intersect = pred_label[pred_label == label]
+    area_intersect = torch.histc(intersect.float(), bins=(num_classes), min=0, max=num_classes - 1)
+    area_pred_label = torch.histc(pred_label.float(), bins=(num_classes), min=0, max=num_classes - 1)
+    area_label = torch.histc(label.float(), bins=(num_classes), min=0, max=num_classes - 1)
+    area_union = area_pred_label + area_label - area_intersect
+
+    return area_intersect, area_union, area_pred_label, area_label
+
+
+def total_intersect_and_union(preds, target, num_classes, ignore_index, label_map=dict(), reduce_zero_label=False):
+    """Calculate Total Intersection and Union.
+
+    Args:
+        preds (list[torch.Tensor]):
+            List of prediction segmentation maps.
+        target (list[torch.Tensor]):
+            List of ground truth segmentation maps.
+        num_classes (int):
+            Number of categories.
+        ignore_index (int):
+            Index that will be ignored in evaluation.
+        label_map (dict):
+            Mapping old labels to new labels. Default: dict().
+        reduce_zero_label (bool):
+            Whether ignore zero label. Default: False.
+
+     Returns:
+        torch.Tensor:
+            The intersection of prediction and ground truth histogram on all classes.
+        torch.Tensor:
+            The union of prediction and ground truth histogram on all classes.
+        torch.Tensor:
+            The prediction histogram on all classes.
+        torch.Tensor:
+            The ground truth histogram on all classes.
+    """
+    total_area_intersect = torch.zeros((num_classes,), dtype=torch.float64)
+    total_area_union = torch.zeros((num_classes,), dtype=torch.float64)
+    total_area_pred_label = torch.zeros((num_classes,), dtype=torch.float64)
+    total_area_label = torch.zeros((num_classes,), dtype=torch.float64)
+
+    for result, gt_seg_map in zip(preds, target):
+        area_intersect, area_union, area_pred_label, area_label = intersect_and_union(
+            result, gt_seg_map, num_classes, ignore_index, label_map, reduce_zero_label
+        )
+
+        total_area_intersect += area_intersect
+        total_area_union += area_union
+        total_area_pred_label += area_pred_label
+        total_area_label += area_label
+
+    return total_area_intersect, total_area_union, total_area_pred_label, total_area_label
+
+
+def _mean_iou_update(
+    preds: Tensor,
+    target: Tensor,
+    num_labels: int,
+    ignore_index: bool,
+    nan_to_num: Optional[int] = None,
+    label_map: Optional[Dict[int, int]] = None,
+    reduce_labels: bool = False,
+) -> Tuple[Tensor, int]:
+    """Updates and returns variables required to compute Mean Intersection over Union.
+
+    Checks for same shape of each element of the ``preds`` and ``target`` lists.
+
+    Args:
+        preds (list[torch.Tensor]):
```

Comment:
remove typehints here as you already added them above. Also: They don't match. The annotations says `torch.Tensor`, the docstring `list[torch.Tensor]`. WHich one is it?